### ISSUE TITLE
Review process for new languages / big updates

### ISSUE THREAD

**Issue by heatherleaf** [GrammaticalFramework/gf-rgl#117]
When someone issues a new Pull Request, there should be an open and transparent review process before it's accepted. I suggest it consists of the following:

1. There must be good documentation: a README and code comments
2. There must be test cases: at least a simple parallel corpus or treebank with 10-20 sentences
3. The PR should be advertised on the mailing list, and people get time to comment on it
4. It should be reviewed by at least 2 people

This is if the PR is a new language or a big update to an existing language. If there is small update (e.g., bug fix or additions to a lexicon), the process can be quicker.

**Comment by heatherleaf** [comment]
If the PR is about a new functor, the process must be even more strict, because functors influence so many languages.

**Comment by johnjcamilleri** [comment]
Good suggestions Peter. These should be placed somewhere official, such as the main README, so that new contributors know what to expect (and for us to refer to).

On this note, I am wary of using PRs a bit too superfluously. Specifically, updating a README like can surely just be committed directly into main by someone with write access.

**Comment by heatherleaf** [comment]
> On this note, I am wary of using PRs a bit too superfluously. Specifically, updating a README like can surely just be committed directly into main by someone with write access.

Of course. I mainly meant that large changes (new languages or big additions or substantial modification) should be PRs, so that it's possible to discuss.

**Comment by odanoburu** [comment]
> so that it's possible to discuss.

not only discuss, but also find out about the changes!
